src/main/java/edu/cornell/kfs/pdp/businessobject/inquiry/CuPaymentDetailInquirable.java

```diff
@@ -0,0 +1,45 @@
+package edu.cornell.kfs.pdp.businessobject.inquiry;
+
+import org.apache.commons.lang.StringUtils;
+import org.kuali.kfs.kns.web.ui.Field;
+import org.kuali.kfs.kns.web.ui.Section;
+import org.kuali.kfs.krad.util.ObjectUtils;
+import org.kuali.kfs.pdp.businessobject.inquiry.PaymentDetailInquirable;
+import org.kuali.rice.location.api.country.Country;
+
+import edu.cornell.kfs.pdp.CUPdpPropertyConstants;
+
+@SuppressWarnings("deprecation")
+public class CuPaymentDetailInquirable extends PaymentDetailInquirable {
+    private static final long serialVersionUID = -5852448786635585758L;
+
+    /**
+     * Overridden to prevent exceptions when the country field is blank,
+     * or when it contains a country name instead of a country code.
+     * 
+     * @see org.kuali.kfs.pdp.businessobject.inquiry.PaymentDetailInquirable#convertCountryForDisplay(org.kuali.kfs.kns.web.ui.Section)
+     */
+    @Override
+    protected void convertCountryForDisplay(Section section) {
+        section.getRows().stream()
+                .flatMap((row) -> row.getFields().stream())
+                .filter(this::isPaymentGroupCountryField)
+                .findFirst()
+                .ifPresent(this::convertCountryForDisplayIfNecessary);
+    }
+
+    protected boolean isPaymentGroupCountryField(Field field) {
+        return StringUtils.equalsIgnoreCase(CUPdpPropertyConstants.PaymentDetail.PAYMENT_COUNTRY, field.getPropertyName());
+    }
+
+    protected void convertCountryForDisplayIfNecessary(Field field) {
+        String propertyValue = field.getPropertyValue();
+        if (StringUtils.isNotBlank(propertyValue)) {
+            Country country = getCountryService().getCountry(propertyValue);
+            if (ObjectUtils.isNotNull(country)) {

```

Comment:
Is it an error condition of a country can't be found?

Reply:
This code is related to just the display of the payment information on an inquiry screen, not its validation. The other processes should be the ones that check for errors when an invalid country is displayed, or if a country code was expected instead of a country name.

We could add a feature that displays an additional error-related message if the country value is a code but doesn't map to an actual country. However, we would need to confirm with the functionals if that's truly the behavior they want, and that would likely be in the scope of a future enhancement unless the functionals say otherwise.

Reply:
OK, I think this implementation is fine then